client/src/providers/User.js

```diff
@@ -0,0 +1,122 @@
+import React, { createContext, useReducer, useEffect, useMemo } from "react";
+import useLocalStorage from "../hooks/useLocalStorage";
+import { gql, useQuery } from "@apollo/client";
+
+const STORE_KEY = "@CashlessUser@1";
+
+const context = {
+  id: null,
+  isAuthenticated: false,
+  secret: {},
+  token: null,
+  profile: {
+    name: null,
+    email: null,
+    description: null,
+  },
+};
+
+const defaultContext = [context, () => {}];
+
+const UserContext = createContext(defaultContext);
+
+function reducer(state, action) {
+  switch (action.type) {
```

Comment:
@jwu910  another stupid question but just wondering - do we really  have to use magic strings here? iS there another pattern?

Reply:
useReducer was a bit newer and these hooks haven't been in the pattern that i had been used to.

I think if anything, if we want to mimic a redux type naming pattern with the context api w/o really using redux, we can just create a an "actions" object for the action types

```javascript
const ACTIONS = {
  SET_SECRET: 'set_secret',
  SE_PROFILE: 'set_profile',
  //...
}
```

And within the switch we can then reference 
```javascript

switch (action.type) {
  case ACTION.SET_SECRET: {}
  case ACTION.SET_PROFILE: {}
}
```

Again, i haven't had an opportunity to use the `useReducer` hook much before I hopped on the vue train, but the code here seems like good dirrection to go.

We spread old state and write in the new data points